components/match2/commons/opponent.lua

```diff
@@ -237,16 +237,25 @@ Resolves the identifiers of an opponent.
 For team opponents, this resolves the team template to a particular date. For
 party opponents, this fills in players' pageNames using their displayNames,
 using data stored in page variables if present.
+
+options.loadFlag: Whether to fetch the flag from variables, match2 or LPDB. Disabled by default
+options.loadTeam: Whether to fetch the team from variables or LPDB. Disabled by default.
 ]]
-function Opponent.resolve(opponent, date)
+function Opponent.resolve(opponent, date, options)
+	options = options or {}
 	if opponent.type == Opponent.team then
 		opponent.template = TeamTemplate.resolve(opponent.template, date) or 'tbd'
 	elseif Opponent.typeIsParty(opponent.type) then
 		local PlayerExt = require('Module:Player/Ext')
 		for _, player in ipairs(opponent.players) do
 			PlayerExt.populatePageName(player)
+			if options.loadFlag then
+				PlayerExt.syncPlayer(player)
+			end
```

Comment:
this change means that #1387 needs an additional function to overwrite this

(needs `Module:Player/Ext/Starcraft` instead of `Module:Player/Ext` plus changing the name of the option)

maybe `options.syncPlayer` is a better name?

Reply:
would it be better to pass the module name to require as an option?
`options.syncPlayer` is the better name for sure, will change.

Reply:
> would it be better to pass the module name to require as an option?

imo no
that remark was more about having this pr show up in the other pr^^
(so i can adjust it once this here is merged)